As a business analyst, it is important to analyze and convey your company's performance using specific metrics in each of the following areas:

* Marketing
* Growth
* Sales
* Finance

<img src="image/business-metrics.png">

* Key Performance Indicators: We will start with a discussion about key performance indicators and how it differs by industries.

* Business Process Flow: Then we will proceed to go through the business process flow across various business divisions. This will provide the context for learning about the business metrics. 

* Business metric: We will take on each business area, such as marketing and growth, and introduce you to a metric commonly used to measure success in that business area. We will discuss what each means, and how to calculate it. We will practice calculating the metrics and applying the metrics, and when and where to use the metric. To do this, we will focus on 3 main elements related to metrics:
Evaluate important business metrics
Interpret and analyze these metrics
Create visualizations of these metrics

* Distribution and central tendency: We will circle back to the topic of data distribution that you learned about in the previous lesson, and why paying attention to the distribution of the data and to the choice of measure of central tendency is important.

* Grouping data: We will end with a discussion on how to look at the data across groups, cohorts, and time.
---

# Key Performance Indicators

Businesses need to be able to track how they are performing on key goals or objectives - whether they are growing number of customers, bringing down their costs, increasing revenue on an ongoing basis, and myriad others.

> Key Performance Indicators or KPIs are how they measure their success on each of their key business objectives.

## Which KPI to use ?

decision regarding which KPI a business analyst should use depends on several factors, including which industry or domain they are working in, which business function they are focusing on, and the type of data they have available to them.

```
https://kpidashboards.com/kpi/
```

### Asking data question

The KPIs you use will be determined by the questions you need to ask. As a business analyst, you are tasked with gathering the appropriate data to help solve business problems. To get to that solution, you will need to:

* Identify what needs to change
* Communicate this change to stakeholders in clear manageable chunks of data

### How to ask questions

1. Identify the business goal and objectives.
2. Narrow down the type of data needed to answer questions.
3. Identify the KPIs that will be useful to show whether you are making progress on your business goal.
4. Conduct the data analysis using the KPIs and use visualizations as part of the analysis.
5. Provide recommendations and findings based on the completed data analysis.
6. Create succinct and visual presentations for the stakeholders.


---

# Business Process Flow

A business has two primary goals:

1. Increase revenues
2. Reduce costs

|Reduce Costs|Increase Revenue|
|---|---|
|Website & Engineering|New Customers|
|Salaries|Marketing|
|Supplies & Payment System||
|Legal Contracts||
|Rent||

Metrics allows businesses to assess whether they are on track to meet these goals

Business executives need to think about the costs related to

* Building
* Promoting
* Operating

## Business Process Flow: Marketing
There are some typical costs involved in the marketing process. Marketing teams incur these costs as they think about how to market their product to their customers.

Marketing is about how and where you get the word out about your product.

Online marketing can take advantage of cookie tracking, which allows customer tracking across time and platforms.

Knowing a customer's online roadmap enables companies to pinpoint places for targeted advertising to these customers and other potential customers like them.

## Business Process Flow: Growth

To grow the business, companies need to not only focus on existing customers, but also on new customers

grow in two ways:

1. Increase new customers
2. Increase the order size and repeated orders of existing customer

---
## Customer Journey
customer's online path to and through their website

 tracking customers through online spaces can take several steps:

1. WeCart can identify specific ad platforms, like search engines and social media sites, to quantify how many people see its ads.
2. When people click on WeCart's ads, they go directly to WeCart's site and a cookie is placed on the viewer's browser. Cookies allow a company to track what site visitors do on its website.
3. The business goal for site visitors may be for visitors to engage with a prompted *call to action* on the website; for example, giving an email address or downloading a brochure.
4. If a visitor, now a potential customer, adds something to a cart, cookies begin tracking their path to becoming a paid WeCart customer.

<img src="image/customersjourney.jpg">

1. Awareness
2. Interest
3. Desire
4. Purchase
5. Post-purchase


> Call To Action (CTA): A marketing term that refers to an action a website visitor is supposed to take when given a specific prompt on a website. These can be words or phrases, or icons that prompt and encourage the user to perform the action.


> Post-Purchase: Actions customers take after purchasing an item that promotes and increase sales and advocate on behalf of the company. For e.g., coming back and purchasing more items, sharing or liking the company or product on social media, taking pictures of the item, and tagging it on Pinterest


---

## Marketing Funnel

marketing funnel is the process of tracking and analyzing each step of the customer journey with data.

### Marketing Funnel Metrics

<img src="image/funnel.jpg">

### 1. Impressions & Reach 
building brand and product awareness using ad platforms and search engine optimization (SEO). SEO allows ads to show up for the right mix of search terms as people search online

    * Impressions – an instance of an advertisement appearing on a website when it is viewed by a visitor.
    
    
### 2. Lead generation 
measures how many visits are made to the website.

    * Click – every time a website visitor views the ad and clicks it
    * Click Thru Rate – number of users that clicked an ad or clicked a link sent via email
    * Cost Per Click
    * Cost Per Lead – indicates a user has become a potential customer or lead because they have expressed interest in the company by downloading a document, creating an account, or providing an email address.
---
#### Click Thru Rate    

As potential customers view the ads, some of those potential customers will click the ad and be taken to the website for the company. To be counted at this level, the user needs to click through the ad

Calculate CTR 

To Calculate WeCart's Click Through Rate (CTR) we need to calculate both the number of impressions and the number of clicks

    * Impressions – record an instance of an advertisement appearing on a website when it is viewed by a visitor. So if you visit the page 4 times, say in one hour, the gross impression count will include each repeated viewing.
    
    * Clicks – every time a website visitor views the ad and clicks it, this gets included in the click count

> Click Through Rate (CTR) = (Clicks/ Impressions) * 100


|Medium|Impressions	|Clicks	|Click Through Rate (%)|
|---|---|---|---|
|Facebook Ad	|1100|	15|	1.36|
|Google Search|	2000|	67	|3.35|
|Google Display	|1500	|25|	1.67|


informs your marketing team whether they should try and increase the number of impressions or when they should reword the ad to increase clicks. Remember, if a person clicks through the ad, it does not mean the customer purchased, but rather they are showing interest in what the ad is about. When your CTR is low, your ad campaign is not generating enough interest. When the CTR increases, it is an indicator of effective and interesting content in your ad campaign, and that maybe you should increase the number of impressions for that ad.


* Click Through Rate (CTR) is the ratio of users clicking on a link or an ad to the number of total users who received the link or saw the ad.
* CTR measures the success of an advertising or email campaign.
* When the CTR increases, it is an indicator of effective and interesting content in your ad campaign, and that maybe you should increase the number of impressions for that ad.
* In general, a 2% CTR is good, however, the rate will vary by industry.

* [Benchmarks for CTRs for Google Ads across industries.](https://www.wordstream.com/blog/ws/2016/02/29/google-adwords-industry-benchmarks)


#### Unique Click Thru Rate    

examined when looking at email campaigns to see how often a link sent through an email was opened by the person receiving the email. If the person receiving the email clicks on the link 5 times, the unique CTR stays one, even though the total CTR is 5. Comparing the unique versus total CTR can help the analyst know if the email campaign reflects the interest among potential customers.
 
1. [Mailigen Website](https://www.mailigen.com/blog/email-campaign-click-through-rate-what-it-is-and-how-to-improve-it/)
    * Discusses how Click Through Rate is one of the main email marketing metrics showing the engagement of your target audience. Also addresses How to Improve It.
2. [Optimizely Website](https://www.optimizely.com/optimization-glossary/click-through-rate/)
    * Discusses how CTR is used to analyze and measure success for emails, webpages, and online advertising.
    
    
#### Cost Per Click

cost to get a click on your ad. It helps us gauge the cost of advertising on the specific platform, so we can see which platform is generating more leads.

Since platforms charge you for the number of ads on a page, you can compare the CPC for the different platforms you are advertising on and see which platform is generating more interactions with your website, or generating more traffic to your website.


$\text{Cost Per Click (CPC)} = \frac{\text{Cost of advertising on source platform}}{\text{number of viewers who clicked on the ad}}$


|Source_platform	|Facebook Ad	|Google Search|	Google Display|
|---|---|---|---|
|Spend|	1,500|	3,000|	5,000|
|Clicks|	700|	2,900|	4,995|
|Cost Per Click (CPC)|	2.14|	1.03|	1.00|


CPC is an indicator of the cost-effectiveness of the ad platform and a useful tool to compare and strategize about which marketing platform is yielding a higher impression and reach and resulting in potential leads.

Different ad platforms cost differently and it is important to remember that while one platform might be cheaper it may not necessarily deliver you as many potential customers as another platform. This is an important trade-off that analysts and marketing teams have to consider.

#### Cost Per Lead

a lead is when a potential customer visits your website and does something on the website in response to a prompt, such as share their email , or download a document, create an account. Once the viewer takes that action, we know the viewer is showing some interest for the product or service, and this could possibly lead to a sale. With Cost Per Lead we are tracking whether the potential customer turned into a lead within a given time period, that could be a 30-day window or 60-day window.

$\text{Cost Per Lead (CPL)} = \frac{\text{Cost of Advertising on Source Platform}}{\text{Total Number of Leads}} $


|Source_platform|	Facebook Ad	|Google Search	|Google Display|
|---|---|---|---|
|Spend|	1,500	|3,000|	5,000|
|Clicks|	700|	2,900|	4,995|
|Cost Per Click (CPC)|	2.14|	1.03|	1.00|
|Leads|	16	|63|	112|
|Cost Per Lead (CPL)|	93.75|	47.62|	44.64|

CPL is an indicator of the cost-effectiveness of the ad platform and a useful tool to compare and strategize about which marketing platforms yielded more leads. A low cost per lead means more of this particular type of person is likely to be interested in the product.

Looking at the data above, we can see that Google Display and Google Ads were comparable in terms of the Cost Per Lead. On the other hand, Facebook was costing us more to get to our potential customers.

* [Marketing Chart](https://www.marketingcharts.com/customer-centric/lead-generation-and-management-79707)
* [Hubspot](https://research.hubspot.com/charts/cost-per-lead-benchmarks-by-industry)

---
### 3. Conversion 
when a lead converts to a paid customer. what the cost is to acquire a paying customer.

   #### Customer Acquisition Cost
    
$\text{Customer Acquisition Cost (CAC)} = \frac{\text{Total sales & marketing costs}}{\text{No. of converted customers}}$    
    
This is the point where a lead, or potential customer, has become a customer by buying something on the website (a product or service). Most companies try to get that number under 25%.

The ultimate goal is to increase the lead-to-customer conversions at the bottom of the funnel. Considering the fact that customer shopping cart abandonment is over 60%, each company's goal is to get higher levels of conversions for the minimum cost of sales and marketing. This leads to the concept of optimizing the marketing funnel.    

|Breakdown|	August|	September	|October|
|---|---|---|---|
|Marketing Costs|	9,500|	12,000|	5,000|
|Sales & Marketing Salaries|	25,000|	25,000	|25,000|
|Overhead costs for Sales and Marketing|	10,000	|8,000	|8,000|
|Total Sales & Marketing Costs|	44,500|	45,000	|38,500|
|Number of Paid Customers	|300	|325	|350|
|Customer Acquisition Cost (CAC)|	148.33|	138.46|	110.00|


   #### Customer Acquisition Cost Method 2
    
 Sometimes it takes a long time for a lead to convert to a customer. For example:

   * A lead may sign up for a free account or download for a few months and then be prompted to become a paying customer then.
   * A marketing campaign may intentionally take some time to realize the revenues it is trying to generate.

To account for this 'lag' in revenue, CAC is often calculated based on a company's average sales cycle. (*averaged across the targeted time period* )

$\text{Customer Acquisition Cost (CAC)} = \frac{\text{Prior Month Marketing Costs + Weight Avg Costs(Overhead + Salaries)}}{\text{No. of converted customers}}$ 



|Breakdown|August|	September|	October|
|---|---|---|---|
|Marketing Costs|	9,500|	12,000	|5,000|
|Sales & Marketing Salaries|	25,000	|25,000	|25,000|
|Overhead costs for Sales and Marketing	|10,000|	8,000|	8,000|
|Number of Paid Customers|	300	|325|	350|
|Weighted Average Costs(60 days/2 months) Prior & Current Month|N/A	|0.5(25,000+10,000)+ 0.5(25,000+8,000)|	0.5(25,000+10,000)+0.5(25,000+8,000)|
|Customer Acquisition Cost (CAC)|	N/A	|133.85|	129.29|

In the example above for CAC is September, since we want to average between August and September, we take 1/2 of the expenses from August and 1/2 from September) and divide that by the # customers in Sept.


The CAC metric is an indicator of how much it costs to acquire a customer. If your customer service team is doing a good job of keeping the paid customers happy, that can lead to future leads and paying customers, and thus keep the cost of acquiring customers low. The company's goal is to keep the CAC low while increasing revenue, as this positively impacts the profit margin and profits.

Spending more than 25% of your revenues means you are spending too much to acquire new customers and spending less indicates that you are losing business opportunities.

---
   #### Conversion Rate

<img src="image/conversion rate.jpg">

$\text{Conversion Rate}=\frac{\text{No. at each level in the funnel }}{\text{No. of Impressions}}$

Optimizing the funnel requires identifying at what level of the funnel your customer loss is the greatest. In other words, are you losing the most customers at the awareness and interest age, or is it when you are converting them into leads?

* If you're losing many of them in the early stages of awareness, you need to focus on the types of ads you're creating, or the ad platforms you're choosing to reach your potential customers.

* If you're losing many of them at the conversion stage, you need to look at your website or online app. It's possible the site is not easy to navigate, and that's why not many customers are converting to paid customers.

So, essentially, you're calculating the success rate at getting a potential customer to do what you want them to do at each level of the marketing funnel, and you compare this number against your impressions.    
    
<img src="image/conversion rate2.jpg"> 



* Conversion rates based on impressions

Basically, at each touchpoint, we'll divide the number of people at each touchpoint divided by the total number of customers.

    * 43 percent roughly of the people who saw the actual ad, downloaded the brochure
    * 19 percent of the people who saw the actual ad arrived on the site, added items to the cart
    * ~8 percent of the people who saw the actual ad actually purchased the item

* Conversion rates based on each touchpoint

Another metric is to calculate the conversion rates based on the Call to Action at the previous level in the funnel, as opposed to the initial impression number as we did above.

So now we divide by the number of people who actually took the call to action of the previous step.

    * 43 percent of that matches what's in the conversion rate based on the impression.
    * 45 percent of those people who downloaded the brochure added their items to the cart.
    * 39 percent of those people actually purchased the item.
    
|Level|	Numbers|Conversion rates based on impression|Conversion rates based on each level|
|---|---|---|---|
|Arrived on site|	1000|||		
|Downloaded brochure|	430|	0.43|	0.43|
|Added item(s) to cart|	193	|0.193|	0.449|
|Purchased item(s)|	75|	0.75|	0.389    |

*Conversion rates based on impressions = Numbers/Arrived on site Numbers

*Conversion rates based on each level = Current level Numbers/Prior Level Numbers

---

   #### Cost Per Acquisition
   
Cost Per Acquisition provides insight into whether or not the marketing campaigns are successful from a business perspective. For the purposes of calculating the CPA, the cost of the marketing campaigns should not be restricted to the cost of developing the ad, but also other costs of labor and overhead. In other words, CPA allows a business to gauge whether the marketing campaign is generating enough potential leads to cover a broader range of costs other than just direct advertisement costs.

$\text{Cost Per Acquisition(CPA) }= \frac{\text{(Marketing and Sales Cost)}}{\text{No. of new leads}}$

|Breakdown|	August|	September|	October|
|---|---|---|---|
|Marketing Costs|	9,500|	12,000	|5,000|
|Sales & Marketing Salaries|	25,000|	25,000|	25,000|
|Overhead costs for Sales and Marketing|	10,000|	8,000|	8,000|
|Total Sales & Marketing Costs	|44,500|	45,000|	38,500|
|Number of Leads (non-paying customers)|	191|	135	|130|
|Cost Per Acquisition (CPA)	|232.98|	333.33|	296.15|

---

  #### CPA VS CAC
  
Cost Per Acquisition (CPA)
* The impact of marketing campaigns can also be measured in terms of revenue using metrics that capture the financial cost.
* Focuses on sales and marketing costs, including the cost of supplies, labor, marketing, overhead, and with a focus on sales leads—not actual paying customers

CAC 
* is focused on actual new customers who have made a purchase
    
### 4. Loyalty

To grow their revenue and company profits, companies don’t just want their customers to buy once from them, but to come back to their website. Especially if the product is not a high-priced product. That customer loyalty allows you to track how many revisits a customer is making after their first purchase, or how many of the customers have continued shopping after their first purchase.

    * Metrics :
        * Repeat Purchase Rate
        * Net Promoter Score
        
### 5. Advocacy

Another level companies sometimes track is whether their customer is advocating for their company. That is, saying good things about the product and services. Leaning on social media provides a great opportunity to do just that.paid customer tweets about the company, likes the product on FB, provides a good rating on Amazon or the company website, analysts can use those metrics, such as ratings and likes to show how many of the customers serve as advocates.

    * Metrics:
        * Customer Referrals
        * Leads from Social Media
        
---

* Search Engine Optimization (SEO): The goal of search engine optimization is to influence the frequency of a website appearing in response to specific search terms in a search engine. You can learn more about it on this Wikipedia page and this [Forbes article](https://www.forbes.com/sites/theyec/2020/03/27/the-10-commandments-of-seo/?sh=43e00e693e07).
* Lead: A potential customer interested in the products or services of a company.
* Conversion: When the lead (potential customer) purchases the products or services being sold by a company.
* Repurchase Rate: This [Medium](https://medium.com/@matsutton/repurchase-rate-the-most-overlooked-ecommerce-kpi-337bccde184b) blog describes how to calculate Repurchase Rate metrics.
* Net Promoter Score: This [Wikipedia](https://en.wikipedia.org/wiki/Net_Promoter) page describes the calculations and origin of NPS.

---

## Lifetime value LTV

When deciding how to spend the marketing budget, you want to focus on some of your best customers – those that will stay for the long-term and continue to generate revenue for the company.

* Purchase Cycle: The time increment adopted for business calculations
* Total Sale Revenue Per Cycle: Revenue earned from a customer per purchase cycle
* Number of Sales Per Purchase Cycle: Number of times customer buys during the purchase cycle
* Cost Per Acquisition: (Cost of marketing and sales)/ number of new leads
* Expected Retention Time: Amount of time (measured in purchasing cycles) you expect to retain the customer.
* Average Sale Revenue: (Total customer revenue/ Number of purchases in the cycle) OR Average revenue received from the customer per transaction during the cycle
* Profit Margin Per Customer: ((Average Sale - Average Cost of Sale) / Average Sale)


* Other methods LTV
    * [Multiple ways to calc LTV](https://neilpatel.com/blog/how-to-calculate-lifetime-value/)
    * [Identify High Value Customers](https://www.intechnic.com/blog/how-to-calculate-customer-lifetime-value-clv-to-market-to-high-value-customers/)
    * [Using cost capital](https://abovecrowd.wordpress.com/2012/09/04/the-dangerous-seduction-of-the-lifetime-value-ltv-formula/)

In [7]:
purchase_cycle = 1 # 1 week
total_sale_revenue_per_cycle = 70 # $ 70 per order x 1 order per week
number_of_sales_per_purchase_cycle = 1 # 1 orders per customer per cycle
cost_per_acquisition = 25
expected_retention_time = 364 # 7 years x 52 weeks
average_sale_revenue = 70 # 70 / 1
profit_margin_per_customer = (average_sale_revenue - cost_per_acquisition) / average_sale_revenue
lifetime_value = average_sale_revenue * number_of_sales_per_purchase_cycle * expected_retention_time * profit_margin_per_customer
print(lifetime_value)

16380.000000000002


In [24]:
purchase_cycle = 2
total_sale_revenue_per_cycle = 8 * 5 
number_of_sales_per_purchase_cycle = 5
cost_per_acquisition = 7.5
expected_retention_time = (3 * 52 )/2
average_sale_revenue = 8
profit_margin_per_customer = (average_sale_revenue - cost_per_acquisition) / average_sale_revenue
lifetime_value = average_sale_revenue * number_of_sales_per_purchase_cycle  * expected_retention_time * profit_margin_per_customer
print(lifetime_value)

195.0


<img src="image/ltv.jpg">

Here is our thinking. Based on the LTV of usd 195, 

we believe Smoothie Rocks should not spend more than usd 195 on the most loyal customers. These customers will continue to bring value to Smoothie Rocks.

If Smoothie Rocks spends more than $195 on their customer, Smoothie Rocks will likely have a loss in the long run because the customers will not stay loyal customers long enough for Smoothie Rocks to recoup the money spent to get these customers.

calculating the LTV across the years, and you're counting the number of transactions by converted customers during each year. For example, you want to know the LTV for a customer acquired in 2016. You want to include all the transactions the customer had even those in 2017. LTV takes into account future uncertainty. It is a good estimate of the future actions of the customer

### Product Value

#### High value
Most high-value products, such as a car or a house, won't have customers coming back and buying them again and again. For example, it's very likely you have bought cars from different dealerships. So, for the dealership, it is more important to focus on the individual customer value rather than the lifetime value. The value you provide each dealership is restricted to that single purchase you made because you never visit them again.

#### Smaller value
Compare that to smaller orders at Amazon. Lifetime value will make sense as you're likely to come back and make orders several times. Calculating lifetime value matters in this case.

## Sale Metrics

* business-to-consumer model or B2C.
focused on the end consumer, and their order as our marketing focus, or unit of analysis.
    
* business-to-business model or B2B
partners with the local grocery stores. We can now gain access to their customers and the local grocery stores can access a delivery service. part of our sales team can focus on the B2B model, where each sales rep is trying to create a sales lead. A sales lead would be a grocery store interested in partnering with We Card. If a grocery store signs a deal, then it becomes a booking or a closed deal.


* Business to Business (B2B): 
When one business makes a business transaction (goods or services) with another business. Often takes place when one business is providing source materials to the other business to in turn finally sell it to the consumer.
* Business to Consumer (B2C): 
When a business sells products and services to the final consumer.

## Sale Funnel
 captures prospects, leads, qualified leads, and bookings at each of the 4 levels in a sales funnel.
 
<img src="image/sales.jpg">

### Sales Pipeline / funnel

tracks the number of incoming leads or prospects. These are leads that a sales team member has identified as being potentially interested in the product. Then, the sales team member follows up for an interested lead to ask more about what they're looking




### Qualified Leads

qualifying process whereas the sales team qualifies the leads, which means they are checking to see if the product offering is within the lead's budget, ultimately in order to identify the ideal buyer and confirm their viable lead.

### Closing deals

Once the sales team has a qualified lead, you end with a closed deal or booking. At this stage, you can also have a lead on hold or last

    * Bookings is a very important metric for tracking the success of the sales team.
    * Close-ratio is the ratio of closed deals to leads from the sales pipeline.
    
    
* Sales Lead: A sales lead refers to the number of potential customers who have shown interest or have been identified by the sales team member as being potentially interested in the product.
* Qualified Lead: A potential lead who has been vetted by the sales team as meeting key requirements of an ideal buyer. Sales teams check to see if the product offering is within the lead’s budget that will make them a viable buyer.
* Booking: Booking is a closed deal when the qualified buyer has committed to making the purchase. It is a key metric for tracking the success of the sales team.
* Sales Pipeline: Refers to the collection of steps a sales representative takes while navigating incoming leads or prospects through to making the final purchase. It is also used to track how well individual sales representatives are meeting their sales quota.    

1. [matrix marketing group](https://www.matrixmarketinggroup.com/sales-pipeline-key-metrics/)
2. [insightsquared](https://www.insightsquared.com/2013/07/mythbusters-what-is-your-ideal-sales-pipeline-to-quota-ratio/)

### Total Bookings

* Booking is a won deal that is signed or where the purchaser is committed to buying the product
* Total Bookings is the sum of all closed deals

### Average Deal Size $

* refers to the average deal size in dollars of all of the won deals. Reminder, a won deal is when the account buyer has committed to making the purchase.

$\text{Average Deal in Size} = \frac{\text{Total Sale Value OR Total Booking}}{\text{No. of Deals/Booking over specific period}}$

* to keep an eye out on the size of the deals you are winning, as any deal that is above this average deal size may involve more risk. The win rate for such sale prospects that are higher than the average deal size is usually low, but that doesn’t mean your sales rep shouldn’t pursue it. Rather these deals should be considered carefully for sales forecasting. Instead, if you see the historical data shows your average deal size is increasing, your sales team can explore and go after lead generation efforts for larger deals. It is also a reminder for the team to understand what is bringing these larger deals into your pipeline.

### Average Time to Close

<img src="image/avgclose.jpg">

* is the average number of days it takes a member of the sales team to close the deal from the prospect stage to a closed deal.
 
The lead source refers to whether the prospect inquired through the website or had an inbound inquiry. On the other hand, outbound methods refer to cold-calling through email lists or phone calls. This means the customer has lower intent to purchase, to begin with, and this lengthens the time to close the deal

* Sum of Total number of days from the first contact to closing the deal for all closed deals
* The average number of days for typical Sales Cycle = Sum(Total number of days to close the deal) for all closed deals / Number of closed deals


In [26]:
(500000 + 850000 + 560000)/3

636666.6666666666

## Growth Metrics

* Are we seeing the number of people actually using the site increasing or decreasing?
* If you see your website use as high, are they unique users or the same people coming back?

### Active Users

* Monthly Active Users: Number of unique active users in the previous month
* Daily Active Users: Number of unique active users the previous day

### Stickiness

stickiness ratio is a very useful tool for several stakeholders, such as investors, to know if the website or app has a potential for growth or either on the trajectory, for growth. Essentially, stickiness tells us if the customers are coming back to the website or app every day, or rather, sticking around to actually engage with it.

* Social networking site is 0.5
* Gaming apps, it is 0.1 to 0.2
* ...most other apps strive to achieve a stickiness ratio of 0.2

$\text{Stickiness Ratio} = \frac{\text{DAU}}{\text{MAU}}$

Investors want to know if this app or website has the potential to make money in the future. For example, if the plan is to introduce advertising into the app, the potential evaluation will depend on whether the app has a large number of users that keep coming back to it.

A higher stickiness ratio indicates the website or online app is engaging the customers.

### Churn Rate

* Churn rate is a measure of declining growth. Business need to make sure that they are acquiring new customers at a rate faster than their "churn rate"
* Churn Rate captures the number of people we retain at the end of a time period.
*  often adopted by companies using a subscriber-based service model, especially in the telecommunication industry. In recent years, e-commerce and SaaS-based companies have adopted the churn rate metric too.
* The churn rate should only tell you whether the current customers have left or stayed.
* To calculate the customer churn rate you need 2 simple things:

    * Customers at the beginning of usage interval
    * Customers at the end of the usage interval

* Customer Churn Rate = (Customers at the beginning of usage interval - Customers at the end of usage interval) / Customers at the beginning of usage interva
* Usage Interval: This time period should make sense for the service or product the customers are using. It can range from a day, a week, to a month or quarter. It depends on the service or product the company is providing and how often you would expect a customer to be active on the website.


For e.g., WeCart is an online grocery business, so we would expect an existing customer to place at least 1 order over a month. We will use the usage interval of 1 month. If the customer has not placed an order over the course of a month, we can count that customer as having churned. Once we have identified these users as having churned, we can focus on the efforts to bring them back to the website and make them active again.

* Annual churn rate of 5% is seen as a reasonable benchmark. Keep in mind that the range for churn rates is wider for B2C companies.
* Select a time interval during which you calculate the churn rate that is consistent with the company's subscription or usage model. There is no ideal usage interval - the usage interval depends on the length of time the company expects the user to be active at least once.
* Pay attention to different customer segments, especially if they have different churn rates (e.g., by region).
* Make sure your data does not include new customers gained during the time interval. Churn rate is focusing on customers who stayed or are active vs. stop being active on the website.

[Chaotic Flow](http://chaotic-flow.com/saas-metrics-faqs-what-is-churn/): Nice article on what data considerations one should keep in mind when calculating the annual churn rate.
[Six Ventures](https://sixteenventures.com/saas-churn-rate): A blog on differentiating between 5% annual vs. 5% monthly churn rate.
[Recurly](https://info.recurly.com/research/churn-rate-benchmarks): This website talks about benchmarks for Churn rate across industries, so check those out to learn about industry-specific churn rates.

## Finance Metrics

Fixed costs remain steady over a period of time and can be anticipated with a fair amount of certainty when running the financial modeling and preparing financial statements. Variable costs change in response to output. To generate more output, products, and services, variable costs relatively grow in proportion.

* [Finance & Accounting](https://corporatefinanceinstitute.com/resources/careers/jobs/finance-vs-accounting/)
* [Careers in Finance](https://www.businessadministrationinformation.com/finance/the-difference-between-finance-and-accounting-degrees)

### Profit & Loss (PNL) or Income Statetement

* Revenues: The money a company makes from the sales of its products and services.
* Cost of Goods Sold (COGS) or Cost of Sales: These are the direct costs the company incurs to develop the product or service being sold.
* Gross Profit: The difference between the revenue earned and the costs summarized in COGS. 
    * Gross Profit = Revenue - COGS
* Selling, General, and Administrative expenses (SGAs): Includes the following expenses:
    * Marketing, sale commissions
    * Salaries for office staff
    * Supplies and computer hardware
    * Note: Some companies list total operating expenses separately from SGAS while others treat them as synonymous with SGAS.
* Operating expenses: Expenses incurred outside of direct manufacturing costs:
    * Overhead costs
    * Legal
    * Rent
    * Utilities
    * Taxes
    * Interest
    * R&D expenses.
* Total Operating Expenses = Sum of SGAs and Operating expenses 
  * Total Operating Expenses= SGAs + Operating Expenses
  * Total Operating Expenses: Expenses incurred outside of direct manufacturing costs
* Operating Income: The difference between Gross profit and Total operating expenses 
  * Operating Income/ Operating Profit/ EBIT = Gross Profit - Total Operating Expenses
  * Operating Income = Gross Profit - Total Operating Expenses
  * Note: Operating Income is also referred to as Earnings Before Interest and Tax (EBIT)
  * Net Income: Subtracting the Interest and Tax from Operating Income gives the Net Income Net Income = Operating Income - (Interest and Taxes)
  * Net Income/ Net Profit = Operating Income - (Interest expense + Tax expense)
  
  
[COGS](https://blog.projectionhub.com/what-is-cost-of-goods-sold-for-a-service-business/)
* Cost of Goods Sold (COGS) doesn’t quite make sense in the service world, as it is only used for product-based businesses. This article looks at a different approach called Cost of Revenue.

### Gross Margin

*  statement about the overall profitability of the company.
* Gross Margin = (Total Sales Revenue – Cost of Goods Sold) / Total Sales Revenue
    * same as for Gross Profit / Total Sales Revenue
    * Gross Margin (in %) = [ (Total Sales Revenue – Cost of Goods Sold) / Total Sales Revenue]*100
    * Gross Margin tells business executives what percentage of each revenue dollar is available to cover operating expenses after the COGS have been accounted for
    
### Fixed Cost
* Fixed costs are expenses that you will incur on a regular, perhaps monthly basis, such as rent, utilities, and employee salaries.
* Fixed costs are also called sunk costs. A good caution to keep in mind is that fixed or sunk costs can increase (for e.g., unexpected rent increases, machinery replacement costs), which is why operational managers prefer the term sunk costs. These sunk costs can prove tricky, because a small increment when taken in bulk, can turn out to be catastrophic for companies, especially start-ups


### Variable Cost
* in response to production output
* cost of materials 
* Variable costs are expenses that move up and down in response to production output. This interpretation is particularly helpful for companies to determine the pricing of the product.


### Contribution Margin Per Unit
* [Contribution Margin](https://www.dummies.com/business/operations-management/how-to-prepare-a-cost-volume-profit-analysis/) tells us the amount of revenue that covers the variable costs and is now available to cover the fixed costs and generate profits. Companies use it to identify which product or product line is contributing the most to the profit margin. It also helps determine the break-even point where the pricing will cover fixed overhead costs and leave enough for profits too.
* which product contribute to most profit margin
* = (Sales Revenue - Variable Cost ) / Total Units Sold 
* P&L statement line items tell us the overall profitability of our business, contribution margin can be used to identify which product or product line is contributing the most to our profit margin.
* If the contribution margin per unit > fixed cost is your profit.
* If the contribution margin per unit < fixed cost, this means you're making a loss on each sale.
* Total Contribution Margin = Total Sales Revenue - Total Variable Cost
* Contribution Margin Per Unit: Total Contribution Margin / Number of Units Sold

<img src="image/margin.jpg">

## Distribution of Data

<img src="image/stats.jpg">

*devil in the detail*

## Forecast
*  go through examples of forecasting sales bookings using both a bottom-up and top-down approach.

### Top Down Approach
* macro approach
*  start with the best estimate of the larger size of the market narrowing it down to identify the portion of the market that the company is serving
* typically adopted when there's limited historical data.

<img src="image/topdown.jpg">

the model is focused on bookings per salesperson and is divided into four sections:

#### Key Seller Assumptions & KPIs

* How Productive will the new salesperson be?
* How many opportunities will be generated?
* How much revenue is generated per unit and per opportunity?

#### Sales Hiring Schedule

* tells us the time and effort needed to generate the expected bookings and opportunities.

#### Sales Productivity Schedule

* Based on historical sales data, we create the assumption that a new sales member would take x number of months to learn the business and "ramp up", in order to generate the bookings and opportunities assumed in the first section.

#### Projected Bookings per Sales Person

* use the "ramp up" time and the assumptions about the average size of opportunities to project out when we will see the bookings that we expect a salesperson to generate.

start with the number of opportunities we expect the salesperson to close annually.

* Average price per unit – the average price per unit or product for manufacturing.
* Average units per opportunity – the average number of units you can expect to sell per opportunity.
* Average contracts month per opportunity – the average length of time in months that sales contract can be for.

> calculate the average opportunity size, you will multiply Average Price Per Unit/month X Average Units per Opportunity X Average Contract Months Per Opportunity.

Next, we get to the average opportunity size, the average booking size we expect this salesperson to create on an annual basis

    * Take the product of the three numbers above to get this number of bookings.
    
Next, we determine how to get to these bookings.

    * Assume that the seller ramp or the length of time we can expect the new seller to reach full productivity after being hired is three months.

The hiring schedule in the model shows that once a person has been hired, a month from then, they will be employed and available to start generating leads.

    * Create dummy coding for the projected hire date and after that, indicate a 0 for not hired and a 1 for a salesperson having been hired. Dummy coding refers to when you use a one and zero as stand-ins or dummies for the presence of something happening.
    
For the center productivity schedule, we want to generate the schedule of when a salesperson will be productive and we want to give that person three months of ramp time.

Finally, we get to our booking projections based on when a seller will be productive.

    * Multiply the productivity dummy variable with the projected average booking and the expected monthly opportunities closed to get the monthly bookings generated by a salesperson.


### Bottom Up Approach
* micro approach
* starts by looking at historical data. The more data you have, the better. But often, even as little as six months to one year of data is used in this approach.

<img src="image/bottomup.jpg">

<img src="image/bottomup2.jpg">


* Contract Terms = Number of months in the contract
* Price per Unit (by mon) = Units needed in 1 month X Price per Unit
* Bookings Forecast = Price per Unit X Contract Term (month)
* Closed/Won Probability = Probability of Closing the deal
* Weighted Bookings Forecast = Bookings Forecast * Closed/Won Probability

### Modelling

#### Inputs
* Historical data : Past Performance
* Assumptions : Changing circumstances
* Scenarios : Possible future

#### Outputs
* Forecasted metrics or KPIs


### Calc Historical Financial Metrics

historical statistics or metrics used to forecast financial metrics in an Income Statement are:

* Revenue Growth
* Gross Margin
* Operating Margin
* Historical Tax Rate
* Historical Interest Expense Rate
* Operating Margin = Current Year's Operating Income / Current Year's Total Revenue
* Historical Tax Rate is the tax rate from the companies previous year's tax rate.
* Historical Interest Rate is the interest rate coming from the previous year's Debt Schedule.

two terms COGS and Cost of Revenue can be used interchangeably.
```
Revenue Growth (in %) = (Current Year's Revenue / Previous year's revenue) - 1
Gross Margin = 1 - (Current Year's Cost of COGS / Current Year's Total Revenue)
```

---

* =INDEX(array, row number, column number)


|Num| A	|B|	C|
|---|---|---|---|
|1	|Fruit|	Color|	Quantity|
|2|	Apple|	Red|	25|
|3|	Banana	|Yellow|	10|
|4	|Grapes|	Green	|7|


*  return the entire third column of the table
INDEX(sample_table,,3) 

* return the value in the first row of the second column.
 INDEX(sample_table,1,2)
 
 Using the simple chart above, for example, if we have an array named fruit_ with the values (Apple, Banana, Grapes) then INDEX(fruit, 3) would return Banana
 
 Note that for the INDEX function, the array could be a single column or the whole table. For example, if the entire table above was named sample_table, then INDEX(sample_table,3) would return the entire third row of the table:

Banana	Yellow	10

* MATCH(Banana, fruit_, 0) returns 3, since the value, Banana, is in row 3 and MATCH gives us the location of a cell within an array.
* INDEX(color_, MATCH(Banana, fruit_, 0)) – returns Yellow, since it is in row 3 of the array color_.
* INDEX(quantity_, MATCH(Banana, fruit_, 0)) – returns 10, since it is in row 3 of the array quantity_.

---

```
* =INDEX(Total_Revenue,1)
* =INDEX(Total_Revenue, MATCH(H2,ticker_symbol,0)) # 1 = greater than , 0 = exact  , -1 = less than
```